torchvision/prototype/models/optical_flow/raft.py

```diff
@@ -0,0 +1,162 @@
+from typing import Optional
+
+from torch.nn.modules.batchnorm import BatchNorm2d
+from torch.nn.modules.instancenorm import InstanceNorm2d
+from torchvision.models.optical_flow import RAFT, BottleneckBlock, ResidualBlock
+from torchvision.models.optical_flow.raft import _raft
+from torchvision.prototype.transforms import RaftEval
+
+from .._api import WeightsEnum, Weights
+
+
+__all__ = (
+    "RAFT",
+    "raft_large",
+    "raft_small",
+)
+
+
+class Raft_Large_Weights(WeightsEnum):
+    C_T = Weights(
+        # Chairs + Things
+        url="",
+        transforms=RaftEval,
+        meta={
+            "recipe": "",
+            "epe": -1234,
+        },
+    )
+
+    C_T_SKHT = Weights(
+        # Chairs + Things + Sintel fine-tuning, i.e.:
+        # Chairs + Things + (Sintel + Kitti + HD1K + Things_clean)
+        # Corresponds to the C+T+S+K+H on paper with fine-tuning on Sintel
+        url="",
+        transforms=RaftEval,
+        meta={
+            "recipe": "",
+            "epe": -1234,
+        },
+    )
+
+    C_T_SKHT_K = Weights(
+        # Chairs + Things + Sintel fine-tuning + Kitti fine-tuning i.e.:
+        # Chairs + Things + (Sintel + Kitti + HD1K + Things_clean) + Kitti
+        # Same as CT_SKHT with extra fine-tuning on Kitti
+        # Corresponds to the C+T+S+K+H on paper with fine-tuning on Sintel and then on Kitti
+        url="",
+        transforms=RaftEval,
+        meta={
+            "recipe": "",
+            "epe": -1234,
+        },
+    )
+
+    default = C_T
+
+
+class Raft_Small_Weights(WeightsEnum):
+    C_T = Weights(
+        url="",  # TODO
+        transforms=RaftEval,
+        meta={
+            "recipe": "",
+            "epe": -1234,
+        },
+    )
+    default = C_T
+
+
+def raft_large(weights: Optional[Raft_Large_Weights] = None, progress=True, **kwargs):
```

Comment:
I'll add the `*`, but regarding the deprecation, I'm not sure we need it.

Since we're planning to move the model prototype into `main` before the next release, users won't ever be using the `pretrained` parameter, so there's no need to handle this.

Reply:
You will need this the moment you put pretrained weights. Some of the tests in prototype still use `pretrained=True` to confirm that the builders between the two areas operate the same using real data. So I would probably recommend adding it here to simplify the situation. You will also plan to for the worse case scenario in which case the prototype builders might not be released in time. Overall I think that RAFT is more likely to be released prior to model builders.